index.js

```diff
@@ -29,18 +28,24 @@ module.exports = function (options) {
 }
 
 async function recurse(base, scope, object) {
+  let i = 0;
   scope = _.clone(scope);
   if (_.isArray(object)) return Promise.all(object.map(_.bind(recurse, this, base, scope)));
   else if (_.isPlainObject(object)) {
     if (object["Fn::Map"]) {
       var args = object["Fn::Map"],
         list = args[0],
         placeholder = args[1],
+        idx = `${placeholder}_IDX`,
```

Comment:
I'm not a fan of special variable names. Instead I'd prefer if the user would specify them:

```
Fn::Map:
  - [1, 2, 3]
  - [Number, Idx]
  - Resource{Number}:
      Type: {Idx}
```